src/main/java/com/google/sps/servlets/TasksServlet.java

```diff
@@ -61,13 +62,76 @@ public TasksServlet(
     this.tasksClientFactory = tasksClientFactory;
   }
 
-  private List<Task> getTasks(TasksClient tasksClient) throws IOException {
+  /**
+   * Returns Tasks from the user's Tasks account
+   *
+   * @param request Http request from client. Should contain idToken and accessToken
+   * @param response 403 if user is not authenticated, list of Tasks otherwise
+   * @param googleCredential a valid google credential object (already verified)
+   * @throws IOException if an issue arises while processing the request
+   */
+  @Override
+  public void doGet(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    assert googleCredential != null
+        : "Null credentials (i.e. unauthenticated requests) should already be handled";
+
+    // Get tasks from Google Tasks
+    TasksClient tasksClient = tasksClientFactory.getTasksClient(googleCredential);
     List<TaskList> taskLists = tasksClient.listTaskLists();
-    List<Task> tasks = new ArrayList<>();
-    for (TaskList taskList : taskLists) {
-      tasks.addAll(tasksClient.listTasks(taskList));
+    List<Task> tasks = getTasks(tasksClient);
+
+    // Initialize Tasks Response
+    List<String> taskListTitles = getTaskListTitles(taskLists);
+    int tasksToComplete = getTasksToComplete(tasks);
+    int tasksDueToday = getTasksDueToday(tasks);
+    int tasksCompletedToday = getTasksCompletedToday(tasks);
+    int tasksOverdue = getTasksOverdue(tasks);
+    TasksResponse tasksResponse =
+        new TasksResponse(
+            taskListTitles, tasksToComplete, tasksDueToday, tasksCompletedToday, tasksOverdue);
+
+    // Convert tasks to JSON and print to response
+    Gson gson = new Gson();
+    String tasksJson = gson.toJson(tasksResponse);
+
+    response.setContentType("application/json");
+    response.getWriter().println(tasksJson);
```

Comment:
This is a good candidate for `JsonUtility`.

Reply:
Agreed - done!